edg/parts/Tle5012b.py

```diff
@@ -0,0 +1,77 @@
+from edg.abstract_parts import *
+from edg.parts.JlcPart import JlcPart
+
+
+class Tle5012b_Device(InternalSubcircuit, FootprintBlock, JlcPart):
+    def __init__(self) -> None:
+        super().__init__()
+        self.vdd = self.Port(VoltageSink(
+            voltage_limits=(3.0, 5.5) * Volt,
+            current_draw=(5 * uAmp, 16 * mAmp)
+        ))
+        self.gnd = self.Port(Ground())
+
+        dio_model = DigitalBidir.from_supply(
+            self.gnd, self.vdd,
+            voltage_limit_tolerance=(-0.3, 0.3) * Volt,
+            input_threshold_factor=(0.3, 0.7),
+            current_limits=(5 * uAmp, 25 * mAmp)
+        )
+        dio_model_sck = DigitalSink.from_supply(self.gnd, self.vdd,
+                                                voltage_limit_tolerance=(-0.3, 0.3) * Volt)
+        self.sck = self.Port(dio_model_sck)
+        self.csq = self.Port(dio_model_sck)
+        self.data = self.Port(DigitalSink.from_bidir(dio_model))
+
+        if_dio_model = DigitalBidir.from_supply(
+            self.gnd, self.vdd,
+            voltage_limit_tolerance=(-0.3, 0.3) * Volt,
+            input_threshold_factor=(0.3, 0.7),
+            current_limits=(5 * uAmp, 15 * mAmp)
+        )
+
+        self.ifa = self.Port(DigitalSource.from_bidir(if_dio_model), optional=True)
+        self.ifb = self.Port(DigitalSource.from_bidir(if_dio_model), optional=True)
+        self.ifc = self.Port(DigitalSource.from_bidir(if_dio_model), optional=True)
+
+    def contents(self) -> None:
+        self.footprint(
+            'U', 'fp:SO8_PG-DSO-8_INF',
```

Comment:
```suggestion
            'U', 'Package_SO:SOP-8_3.9x4.9mm_P1.27mm',
```
Can a standard KiCad footprint be used? This one should be compatible, though the pads stretch more on the inside